tsdb/engine/tsm1/engine.go

```diff
@@ -184,6 +185,9 @@ func (e *DevEngine) LoadMetadataIndex(shard *tsdb.Shard, index *tsdb.DatabaseInd
 // WritePoints writes metadata and point data into the engine.
 // Returns an error if new points are added to an existing key.
 func (e *DevEngine) WritePoints(points []models.Point, measurementFieldsToSave map[string]*tsdb.MeasurementFields, seriesToCreate []*tsdb.SeriesCreate) error {
+	e.mu.RLock()
```

Comment:
This should go after the creation of the values map from points to reduce lock contention.  e.g. Just before L199